contracts/StarStaking.sol

```diff
@@ -273,35 +283,52 @@ contract StarStaking is StarStakingInterface, Lockable {
 
         while (
             (node != HEAD) && ((totalStakingPointsFor[node] < _stakingPoints))
-            || (node == msg.sender)
+            || (node == _user)
         ) {
             node = getTopRank(node, PREV);
         }
 
         if (node == HEAD) {
             node = getTopRank(HEAD, NEXT);
 
-            if (node == msg.sender) node = getTopRank(node, NEXT);
+            if (node == _user) node = getTopRank(node, NEXT);
         }
 
         return node;
     }
 
     /**
-     * @dev Reads the current discount in % for the passed staking points.
-     * @param _stakingPoints The staking points to be used.
-     * @return The discount.for given staking points.
+     * @dev Reads the current discount in per mill for the passed staking points
+     * @param _stakingPoints The staking points to be used
+     * @param _user the user address
+     * @return The discount for given staking points
      */
-    function getDiscountEstimateForPoints(uint256 _stakingPoints)
+    function getDiscountEstimateForPointsForUser(
+        uint256 _stakingPoints,
+        address _user
+    )
         public
         view
         returns (uint256)
     {
-        address oneRankAbove = getSortedSpotForPoints(_stakingPoints);
-        address replacedUser = getTopRank(oneRankAbove, NEXT);
+        if (topRanksCount == 0) {
+            return _computeDiscountForRank(0);
+        }
+
+        address notInTopRanks = 0xd20B0A19D1806f4f6F5a714EddF8e3e9807e2d9f;
```

Comment:
Why the hard coded address? 
It is prone to be brittle.

Reply:
I know this is somewhat of a hack. The reason is because `getSortedSpotForPointsForUser` returns (except for first rank) oneRankAbove for the user which may or may not be already in the list. So just to be sure we are receiving the proper oneRankAbove, I pass an address that is guaranteed to not be in the list.

- I could either refactor `getSortedSpotForPointsForUser` to address this OR
- add a comment

WDYT? I think refactoring would be somewhat difficult, maybe a comment is sufficient?

Reply:
It is indeed a hack. If this needs to be pushed soon I would add a comment then